main.tf

```diff
@@ -1,3 +1,13 @@
+resource "terraform_data" "validator" {
+  // https://github.com/hashicorp/terraform/issues/31122
+  lifecycle {
+    precondition {
+        condition = !(var.redirect_http_to_https == true && var.ssl_certificate_enabled == false)
+        error_message = "redirect_http_to_https can only be enabled when ssl_certificate_id is set to a valid value"
```

Comment:
You can't make variable validation reference other variables. The link in the comment talks about this limitation in Terraform and offers this is a workaround.

Reply:
That's a neat workaround - the other option might be to just have a local that computes that boolean logic and uses it for the conditionals further down?  It won't error out but it will only deploy if both these conditions are met?

Reply:
That's a nice solution, I've changed it as suggested.